bin/accessibility-automation/helper.js

```diff
@@ -0,0 +1,219 @@
+      );
+    } else {
+      if(error.message === 'Invalid configuration passed.') {
+        logger.error(
+          `Exception while creating test run for BrowserStack Accessibility Automation: ${
+            error.message || error.stack
+          }`
+        );
+        for(const errorkey of error.errors){
+          logger.error(errorkey.message);
+        }
+        
+      } else {
+        logger.error(
+          `Exception while creating test run for BrowserStack Accessibility Automation: ${
+            error.message || error.stack
+          }`
+        );
+      }
+      // since create accessibility session failed
+      process.env.BROWSERSTACK_TEST_ACCESSIBILITY = 'false';
+      user_config.run_settings.accessibility = false; 
+    }
+  }
+}
+
+const nodeRequest = (type, url, data, config) => {
+  logger.info("API URL IN noderequest", API_URL);
+  return new Promise(async (resolve, reject) => {
+    const options = {...config,...{
+      method: type,
+      url: `${API_URL}/${url}`,
+      body: data,
+      json: config.headers['Content-Type'] === 'application/json',
+    }};
+
+    request(options, function callback(error, response, body) {
+      if(error) {
+        logger.info("error in nodeRequest", error);
+        reject(error);
+      } else if(!(response.statusCode == 201 || response.statusCode == 200)) {
+        logger.info("response.statusCode in nodeRequest", response.statusCode);
+        reject(response && response.body ? response.body : `Received response from BrowserStack Server with status : ${response.statusCode}`);
+      } else {
+        try {
+          if(typeof(body) !== 'object') body = JSON.parse(body);
+        } catch(e) {
+          if(!url.includes('/stop')) {
+            reject('Not a JSON response from BrowserStack Server');
+          }
+        }
+        resolve({
+          data: body
+        });
+      }
+    });
+  });
+}
+
+const getBuildDetails = (bsConfig) => {
+  let buildName = '',
+      projectName = '',
+      buildDescription = '',
+      buildTags = [];
+  
+  /* Pick from environment variables */
+  buildName = process.env.BROWSERSTACK_BUILD_NAME || buildName;
+  projectName = process.env.BROWSERSTACK_PROJECT_NAME || projectName;
+  
+  /* Pick from run settings */
+  buildName = buildName || bsConfig["run_settings"]["build_name"];
+  projectName = projectName || bsConfig["run_settings"]["project_name"];
+  if(!utils.isUndefined(bsConfig["run_settings"]["build_tag"])) buildTags = [...buildTags, bsConfig["run_settings"]["build_tag"]];
+
+  buildName = buildName || path.basename(path.resolve(process.cwd()));
+
+  return {
+    buildName,
+    projectName,
+    buildDescription,
+    buildTags
+  };
+}
+
+const getAccessibilityCypressCommandEventListener = () => {
+  return (
+    `require('browserstack-cypress-cli/bin/accessibility-automation/cypress');`
+  );
+}
+
+const setAccessibilityEventListeners = () => {
```

Comment:
Use common method already used by O11y by parameterizing it

Reply:
Will refactor in v2 pr.